src/settings/SettingsModal.tsx

```diff
@@ -101,20 +104,36 @@ export const SettingsModal: FC<Props> = ({
   const [showDeveloperSettingsTab] = useSetting(developerMode);
 
   const { available: isRageshakeAvailable } = useSubmitRageshake();
+  // If we are on ios we will show a button to open the native picker.
+  const iosDeviceMenu = useObservableEagerState(iosDeviceMenu$);
+  // In controlled devices we will not show the input section
```

Comment:
How come? Wouldn't the `iosDeviceMenu` condition be more appropriate here?

Reply:
There are 3 cases:
 - normal (input:yes + output:yes)
 - ios (input:no, output: menu button + earpice picker vs speaker in output)
 - android (input: no ouput:yes)

 we model them with the two bools iosDeviceMenu, controlledMediaDevices. where one is responsible for the input: yes/no and the other on for output: default/custom button ios thing

Reply:
The controlled media devices bool value is responsible to control whether the input is shown. If it is true we do not show the input if it is false we do show the input selection.

So yes in controller media devices input is never shown and that is how/why they define the input visibilty

Reply:
On Android we hide the input because it also uses controlled media devices and the os does not expose input selection. It combines a mic and a speaker into one media device. So we change both at once and output is easier to understand. (Main speaker vs bluetooth device in input is confusing. But in output it makes a lot of sense and that the input is switched at the same time is hopefully expected and hence not too confusing for the user.)

Reply:
I think I get it now: for the use cases in which `mediaDeviceControls` would be active, we actually expect both the audio input devices and audio output devices to reflect the very same device list? If so, I would suggest renaming the URL parameter to `audioDeviceControls` and likewise changing the `controls` API names & docs to reference "audio input/output devices" (or maybe just "audio devices") as a unified concept, rather than specifically "audio output devices".

I also suggest you add a code comment with the explanation you just gave for why the devices are easier to understand as output devices, in the UI - I found that helpful.

Reply:
The rename also makes sense but I dont see to much value in the output vs media choice. Media is very vague so it could also contain input (and even video tbh) so I think it is acceptable.
We also have rust and mobile code that already uses the naming: `mediaDeviceControls`

Reply:
Ah right, `controlledAudioDevices` was what I meant to suggest.

> Media is very vague so it could also contain input (and even video tbh)

That last bit, that it could apply to video devices, is how I see `controlledAudioDevices` could be an improvement. How set-in-stone is the Rust and mobile code?

With the other names, I kind of just find them misleading if they only mention outputs. IMO we should make their usage expectations (that input + output are coupled) be reflected in their names while we have the chance:

```typescript
setAvailableAudioDevices(devices: AudioDevice[]): void;
setAudioDevice(id: string): void;
onAudioDeviceSelect?: (id: string) => void;
setAudioEnabled(enabled: boolean): void;
showNativeAudioDevicePicker?: () => void;
```